perception/autoware_tensorrt_bevdet/include/autoware/tensorrt_bevdet/bevdet_node.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef AUTOWARE__TENSORRT_BEVDET__BEVDET_NODE_HPP_
+#define AUTOWARE__TENSORRT_BEVDET__BEVDET_NODE_HPP_
+
+#include "bevdet.hpp"
+#include "cpu_jpegdecoder.hpp"
+
+#include <opencv2/opencv.hpp>
+
+#include <autoware_perception_msgs/msg/detected_object_kinematics.hpp>
+#include <autoware_perception_msgs/msg/detected_objects.hpp>
+#include <autoware_perception_msgs/msg/object_classification.hpp>
+#include <autoware_perception_msgs/msg/shape.hpp>
+#include <geometry_msgs/msg/point32.hpp>
+#include <geometry_msgs/msg/quaternion.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <sensor_msgs/msg/point_cloud2.hpp>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+
+#include <cv_bridge/cv_bridge.h>
+#include <message_filters/subscriber.h>
+#include <message_filters/sync_policies/approximate_time.h>
+#include <message_filters/synchronizer.h>
+#include <pcl/common/transforms.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>  // msg2pcl
+#include <yaml-cpp/yaml.h>
+
+#include <cassert>
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+#include <thread>
+#include <vector>
+
+uint8_t getSemanticType(const std::string & class_name);
+void Getinfo(void);
+
+void box3DToDetectedObjects(
+  const std::vector<Box> & boxes, autoware_perception_msgs::msg::DetectedObjects & objects,
+  const std::vector<std::string> & class_names, float score_thre, const bool has_twist);
+
+// opencv Mat-> std::vector<char>
+int cvToArr(cv::Mat img, std::vector<char> & raw_data)
+{
+  if (img.empty()) {
+    std::cerr << "image is empty. " << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  std::vector<u_char> raw_data_;
+  cv::imencode(".jpg", img, raw_data_);
+  raw_data = std::vector<char>(raw_data_.begin(), raw_data_.end());
+  return EXIT_SUCCESS;
+}
+
+int cvImgToArr(std::vector<cv::Mat> & imgs, std::vector<std::vector<char>> & imgs_data)
+{
+  imgs_data.resize(imgs.size());
+
+  for (size_t i = 0; i < imgs_data.size(); i++) {
+    if (cvToArr(imgs[i], imgs_data[i])) {
+      return EXIT_FAILURE;
+    }
+  }
+  return EXIT_SUCCESS;
+}
+
+class TRTBEVDetNode : public rclcpp::Node
+{
+private:
+  size_t img_N_;
+  int img_w_;
```

Comment:
@cyn-liu 

Thank you for your PR!
I'd like to provide some comments for improvement. Although int is frequently used in Autoware, it is actualy an unsafe type. Please use uint8_t instaed.
cf. https://matt.sh/howto-c#:~:text=Writing%20code-,Types,-If%20you%20find

Reply:
`uint8_t` cannot represent the range of 1600.

Reply:
So please use `uint32_t` instead